src/utils/mod.rs

```diff
@@ -1,28 +1,20 @@
 //! Utilities used in this crate which can also be generally useful downstream.
 
+use alloc::string::String;
 use core::fmt::{self, Display, Write};
 
-#[cfg(feature = "std")]
-pub use std::{format, vec};
-
-#[cfg(not(feature = "std"))]
-pub use alloc::{format, vec};
-
 use super::Word;
-use crate::utils::string::*;
 
 mod kv_map;
 
 // RE-EXPORTS
 // ================================================================================================
 pub use winter_utils::{
-    boxed, string, uninit_vector, Box, ByteReader, ByteWriter, Deserializable,
-    DeserializationError, Serializable, SliceReader,
+    uninit_vector, Box, ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable,
+    SliceReader,
 };
```

Comment:
Since we are removing `utils::boxed`, `utils::string`, and a few other items from the public interface, would this not be a breaking change?

If so, I think this PR should probably go against `next`. Or if we keep the re-exports and add deprecation warnings, then this could go against `main`.

Reply:
Ah, good point, I'll add those back with deprecation warnings (I actually can't recall if the deprecation warnings from `winter-utils` propagate or not, I think they do, but I'll try to confirm quick).

Reply:
I've added those back, and it looks like by re-exporting them compiling `miden-crypto` will produce compilation warnings for referencing those modules, which is fine. I did look to see if I could silence them when building `miden-crypto`, but produce new deprecation warnings if the re-exported modules are referenced, but unfortunately that doesn't appear to be possible (or at least my little test couldn't trigger the deprecation warning in that configuration).

Anyway, should be good to go now, but just wanted to note that one detail.

Reply:
Thank you! Yes, this should work fine - but I'm not seeing any new commits on this branch.

Reply:
I force pushed

Reply:
Actually, looks like I accidentally pushed to `origin` rather than my fork, but it obviously succeeded since you added me as a contributor 😅. I pushed to the right origin this time, should be good to go now.